packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/bedrock/invoke-model.ts

```diff
@@ -124,12 +124,14 @@ export class BedrockInvokeModel extends sfn.TaskStateBase {
 
   constructor(scope: Construct, id: string, private readonly props: BedrockInvokeModelProps) {
     super(scope, id, props);
+
     this.integrationPattern = props.integrationPattern ?? sfn.IntegrationPattern.REQUEST_RESPONSE;
 
     validatePatternSupported(this.integrationPattern, BedrockInvokeModel.SUPPORTED_INTEGRATION_PATTERNS);
 
     const isBodySpecified = props.body !== undefined;
-    const isInputSpecified = props.input !== undefined && props.input.s3Location !== undefined;
+    //Either specific props.input with bucket name and object key or input s3 path
+    const isInputSpecified = (props.input !== undefined && props.input.s3Location !== undefined) || (props.inputPath !== undefined);
```

Comment:
if `props.input != undefined` is true why do we need to check for `props.input.s3Location` (should it be `||` there instead of `&&`)?

Reply:
This is how customer can specify an input : 
1. either using props.input, if they are using this then input.s3Location is required which is defined under the input type (BedrockInvokeModelInputProps), so we need to check for both, hence an `AND` condition. 

2. use inputPath in which case input.props is not needed 
Result should be either of this condition, hence an or